test/vfd.c

```diff
@@ -4911,13 +4922,13 @@ test_vector_io(const char *vfd_name)
 
     for (i = 0; i < count; i++) {
 
-        HDfree(write_bufs_0[i]);
+        HDfree((void *)write_bufs_0[i]);
```

Comment:
Since H5FDwrite_vector takes a const void ** argument, the cleanest way to fix the warnings in this file is to have them actually be const void **. Unfortunately that means we will always have const warnings from freeing those pointers, unless we use our flex_ptr union hack here.